CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+find_package(Git REQUIRED QUIET)
```

Comment:
Could you explain why ? I agree that FetchContent need it so it should be safe but it should not hurt just for safety / be clear to the user that it is needed

Reply:
Maybe you can FetchContent from tarball release on systems without git (maybe a lightweight docker)

Reply:
from CMake doc:

The `QUIET` option disables informational messages, including those indicating that the package cannot be found if it is not `REQUIRED`. 
### The `REQUIRED` option stops processing with an error message if the package cannot be found.

Reply:
That is the needed behaviour, Git is needed to extract the version

Reply:
> Maybe you can FetchContent from tarball release on systems without git (maybe a lightweight docker)

I agree but without Git the purpose of CPM is very limited. In this case people could still use get_cpm. My idea was to provide an alternative to get_cpm. If we want to get ou of Git this requires more changes (Maybe in next PR :) )

Reply:
I think it is still useful as is because the user does not have to include an external file of `get_cpm` and instead use more native commands cmake commands.